filtercascade/__init__.py

```diff
@@ -113,22 +156,49 @@ def from_buf(cls, buf):
 
 
 class FilterCascade:
-    DIFF_FMT = b"<III"
-    VERSION_FMT = b"<H"
+    # The metadata struct is three values defining the configuration settings
+    # used in the production of the filter cascade, per-level.
+    # bytes 0-3: filtersize in bits as an unsigned int
+    # bytes 4-7: number of hash functions for this level, as an unsigned int
+    # bytes 8-11: level number, as an unsigned int
+    diff_struct = struct.Struct(b"<III")
+
+    # The version struct is a simple 2-byte short indicating version number
+    # bytes 0-1: The version number of this filter, as an unsigned short
+    version_struct = struct.Struct(b"<H")
+
+    # version 2 filters, after the version_struct field, have
+    # the following:
+    # byte 0: Whether the decision logic should be inverted, as a boolean
+    # byte 1: Hash algorithm enum per HashAlgorithm, as an unsigned char
+    # byte 2: L, length of salt field as a unsigned char
+    # bytes 3+: salt field of length L
+    version_2_salt_struct = struct.Struct(b"<?BB")
 
     def __init__(
         self,
         filters,
         error_rates=[0.02, 0.5],
         growth_factor=1.1,
         min_filter_length=10000,
-        version=1,
+        version=2,
+        hashAlg=HashAlgorithm.MURMUR3,
+        salt=None,
     ):
         self.filters = filters
         self.error_rates = error_rates
         self.growth_factor = growth_factor
         self.min_filter_length = min_filter_length
         self.version = version
+        self.hashAlg = hashAlg
+        self.salt = salt
+
+        if self.salt and version < 2:
+            raise ValueError("salts require format version 2 or greater")
+        if self.salt and not isinstance(self.salt, bytes):
+            raise ValueError("salts must be passed as byteas")
+        if self.salt and self.hashAlg == HashAlgorithm.MURMUR3:
+            raise ValueError("salts not permitted for MurmurHash3")
```

Comment:
Add a check for using SHA-256 with version 1? Also maybe we should require a salt when using SHA-256?

Reply:
Good ideas.

Reply:
I don't think it should require a salt for SHA256, for what it's worth. Murmur is unsalted, after all.

Reply:
(This would be easy to change in the future.)